EasyPost.Tests/ServicesTests/Beta/RateServiceTest.cs

```diff
@@ -0,0 +1,85 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using EasyPost.Calculation;
+using EasyPost.Models.API.Beta;
+using EasyPost.Tests._Utilities;
+using EasyPost.Tests._Utilities.Annotations;
+using EasyPost.Utilities.Annotations;
+using Xunit;
+
+namespace EasyPost.Tests.ServicesTests.Beta
+{
+    public class RateServiceTests : UnitTest
+    {
+        public RateServiceTests() : base("beta_rate_service")
+        {
+        }
+
+        #region Tests
+
+        #region Test CRUD Operations
+
+        [Fact]
+        [CrudOperations.Read]
+        [Testing.Function]
+        public async Task TestRetrieve()
+        {
+            UseVCR("retrieve");
+
+            Dictionary<string, object> data = Fixtures.BasicShipment;
+
+            List<StatelessRate> rates = await Client.Beta.Rate.RetrieveStatelessRates(data);
+
+            Assert.IsType<List<StatelessRate>>(rates);
+        }
+
+        #endregion
+
+        [Fact]
+        [Testing.Function]
+        public async Task TestGetLowestRateInstanceFunction()
+        {
+            UseVCR("get_lowest_rate_instance_function");
+
+            Dictionary<string, object> data = Fixtures.BasicShipment;
+
+            List<StatelessRate> rates = await Client.Beta.Rate.RetrieveStatelessRates(data);
+
+            StatelessRate lowestStatelessRate = Calculation.Rates.GetLowestStatelessRate(rates);
+
+            Assert.Equal("First", lowestStatelessRate.Service);
+        }
+
+        [Fact]
+        [Testing.Function]
+        public async Task TestGetLowestRateStaticFunction()
+        {
+            UseVCR("get_lowest_rate_static_function");
+
+            Dictionary<string, object> data = Fixtures.BasicShipment;
+
+            List<StatelessRate> rates = await Client.Beta.Rate.RetrieveStatelessRates(data);
+
+            StatelessRate lowestStatelessRate = Client.Beta.Rate.GetLowestStatelessRate(rates);
+
+            Assert.Equal("First", lowestStatelessRate.Service);
+        }
+
+        [Fact]
+        [Testing.Function]
+        public async Task TestGetLowestRateExtensionFunction()
+        {
+            UseVCR("get_lowest_rate_extension_function");
+
+            Dictionary<string, object> data = Fixtures.BasicShipment;
+
+            List<StatelessRate> rates = await Client.Beta.Rate.RetrieveStatelessRates(data);
+
+            StatelessRate lowestStatelessRate = rates.GetLowest();
```

Comment:
How does this even work?...

Reply:
Extension method magic in C#, basically an method that's available as an instance on a particular type (peep the `this` keyword on the first parameter):
https://github.com/EasyPost/easypost-csharp/blob/71e732e48c3e6ea257c956628dcf33b3bef9907f/EasyPost/Calculation/Rates.cs#L127